src/plugins/ui_stateful/clipboard.ts

```diff
@@ -517,11 +553,31 @@ export class ClipboardPlugin extends UIPlugin {
     );
   }
 
-  private getHTMLContent(): string {
+  private async getHTMLContent(): Promise<string> {
     let innerHTML: string = "";
     const cells = this.copiedData?.cells;
     if (!cells) {
-      innerHTML = "\t";
+      if (this.copiedData?.figureId) {
+        const figureId = this.copiedData.figureId;
+        const figureSheetId = this.getters.getFigureSheetId(figureId)!;
+        const figure = this.getters.getFigure(figureSheetId, figureId)!;
+        if (figure.tag == "image") {
+          if (!this.fileStore) {
+            return "\t";
+          }
+          const imageUrl = this.getters.getImage(figureId).path;
+          const file = (await this.fileStore?.getFile(imageUrl)) || null;
+
+          if (file) {
+            const imageUrl = (await this.readFileAsDataURL(file)) as string;
+            innerHTML = `<img src="${xmlEscape(imageUrl)}" />`;
```

Comment:
most of the time, it'll probably be an actual public url, easier than encoding everything. Such an approach would require for the integrator to develop a "getPublicUrl" method to expose attachments and such to public requests. Technically feasible but it makes the feature hard to use in stand-alone, which is what I was trying to avoid.

Reply:
I realized I didn't directly answer the question but yes, if you get to inspect your clipboard after copying an image you'll have an html structure similar to this one, with an src that points to the file or an encoded one (see what I explain above). I didn't find the spec about this but I know that some programs (ms 365 suite for instance) solely rely on that information and completely ignore the DataTransfer items attached in the clipboard. You can test it out by copying an image in your PC and try to paste it in Word online.
<img width="399" alt="image" src="https://github.com/user-attachments/assets/2beee924-8bdf-44d6-9d88-582fd08a0c93" />